### ISSUE TITLE
Learning rate change

### ISSUE THREAD

**Issue by david-bernstein** [wohlert/generative-query-network-pytorch#5]
Regarding line 113 of run-gqn.py. Does this change the learning rate of the Adam optimizer? This post shows something different

https://stackoverflow.com/questions/48324152/pytorch-how-to-change-the-learning-rate-of-an-optimizer-at-any-given-moment-no

**Comment by wohlert** [comment]
Thanks Dave, I have updated the code to accommodate this change.

**Comment by Paandaman** [comment]
Just another question regarding the learning rate and pixel variance: are they not initialized incorrectly? We've got: 
     
    # Pixel variance
    sigma_f, sigma_i = 0.7, 2.0
    # Learning rate
    mu_f, mu_i = 5*10**(-5), 5*10**(-4)
    mu, sigma = mu_f, sigma_f
So mu starts of as 5*10**(-5) but after the first batch it starts of decaying from the value mu_i so there's a leap in the value. Same goes for sigma which now starts at 0.7 but after the first batch is decaying from sigma_i so from 2.0. Please correct me if I'm misunderstanding something! If I'm correct then this would only result in a slight jump in the values after the first batch but after that it should work as intended.

**Comment by wohlert** [comment]
This has already been discussed in [issue 6](https://github.com/wohlert/generative-query-network-pytorch/issues/6). You are correct, but there appears to be no real difference after the first batch. I will change it next time there is some other big error.

**Comment by Paandaman** [comment]
Oh sorry, I was only looking at the open issues and missed that it had already been discussed! Agreed, it doesn't seem to have any significant effect.
